src/vdom/component.js

```diff
@@ -75,8 +75,12 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		initialBase = isUpdate || nextBase,
 		initialChildComponent = component._component,
 		skip = false,
+		snapshot = previousContext,
 		rendered, inst, cbase;
 
+	// FIXME: Do we have an issue with previous state??
+	const prev = extend({}, previousState);
```

Comment:
Dirty workaround: I have the suspicion that previousState is mutated somewhere. Caught this with the tests for getSnapshotBeforeUpdate by react.

TODO: Check what's happening here. Seems fishy.